plugins/org.yakindu.sct.generator.cpp/src/org/yakindu/sct/generator/cpp/CppCodeGeneratorModule.java

```diff
@@ -61,11 +65,22 @@ public void configure(GeneratorEntry entry, Binder binder) {
 		binder.bind(INamingService.class).to(CppNamingService.class);
 		binder.bind(ITypeSystemInferrer.class).to(STextTypeInferrer.class);
 		binder.bind(Naming.class).to(CppNaming.class);
+		bindTracingProperty(entry, binder);
 		bindEventDrivenClasses(entry, binder);
 		bindIGenArtifactConfigurations(entry, binder);
 		addIncludeProvider(binder, ScTypesIncludeProvider.class);
 		addIncludeProvider(binder, CppInterfaceIncludeProvider.class);
 	}
+	
+	protected void bindTracingProperty(GeneratorEntry entry, Binder binder) {
+		FeatureParameterValue traceEnterFeature = entry.getFeatureParameterValue(FEATURE_TRACING,
+				PARAMETER_TRACING_ENTER_STATE);
+		FeatureParameterValue traceExitFeature = entry.getFeatureParameterValue(FEATURE_TRACING,
+				PARAMETER_TRACING_EXIT_STATE);
+		boolean traceEnter = traceEnterFeature != null ? traceEnterFeature.getBooleanValue() : false;
+		boolean traceExit = traceExitFeature != null ? traceEnterFeature.getBooleanValue() : false;
+		binder.bind(Boolean.class).annotatedWith(Names.named(ADD_TRACES)).toInstance(traceEnter || traceExit);
```

Comment:
Why is this needed additionally to `GenModelEntries.tracingUsed()`?

Reply:
I'm sorry. Now I got what you mean.

I could inject GenmodelEntries ``@Inject GenmodelEntries entries;``

and replace the whole thing with 
``binder.bind(Boolean.class).annotatedWith(Names.named(ADD_TRACES)).toInstance(entries.tracingUsed(entry));``

Reply:
And I cannot use ``GenModelEntries.tracingUsed(entry)``, because the functions are not static